torch/distributed/_tensor/device_mesh.py

```diff
@@ -172,8 +171,12 @@ def _init_process_groups(self):
         # validate that all calling ranks pass in the same `mesh` argument.
         self_mesh = self.mesh.to(self.device_type)
         mesh_list = [self_mesh.clone() for _ in range(get_world_size())]
-        all_gather(mesh_list, self_mesh)
-        for other_rank, other_mesh in enumerate(mesh_list):
+        mesh_tensor = torch.cat(mesh_list)
+        funcol.all_gather_tensor(mesh_tensor, gather_dim=0, group=_get_default_group())
```

Comment:
you must to use the result of `all_gather_tensor` as functional collectives are not in-place.

Reply:
yes good catch. thats hard to remember from looking at the apis :/ can we improve somehow?

Reply:
We plan to remove the `_into_tensor` and `_tensor` suffixes that we think are confusing.